subsys/bluetooth/mesh/adv.h

```diff
@@ -75,18 +77,18 @@ int bt_mesh_adv_start(const struct bt_le_adv_param *param, int32_t duration,
 		      const struct bt_data *sd, size_t sd_len);
 
 static inline void bt_mesh_adv_send_start(uint16_t duration, int err,
-					  const struct bt_mesh_send_cb *cb,
-					  void *cb_data)
+					  struct bt_mesh_adv const *adv)
 {
-	if (cb && cb->start) {
-		cb->start(duration, err, cb_data);
+	if (adv->cb && adv->cb->start && !adv->started) {
+		adv->started = 1;
+		cb->start(duration, err, adv->cb_data);
 	}
 }
 
 static inline void bt_mesh_adv_send_end(
-	int err, const struct bt_mesh_send_cb *cb, void *cb_data)
+	int err, struct bt_mesh_adv const *adv)
 {
-	if (cb && cb->end) {
-		cb->end(err, cb_data);
+	if (adv->cb && adv->cb->end) {
```

Comment:
```suggestion
	if (adv->cb && adv->cb->end && adv->started) {
```

Reply:
I don’t think so, some `cb_start=NULL`, just want `cb_end`, so `adv->started` will be `false`.

Reply:
https://github.com/zephyrproject-rtos/zephyr/blob/cd465ff8c4a606209e829170f775791ef475acf9/subsys/bluetooth/mesh/friend.c#L770-L773